### ISSUE TITLE
switching from npm dep to a github dep requires a cache clean

### ISSUE THREAD

**Issue by tamird** [yarnpkg/yarn#1893]
**What is the current behavior?**

Given the following `package.json`:
```
{
  "dependencies": {
    "jspm": "^0.16.33"
  }
}
```
running `yarn install` correctly installs `jspm` into `node_modules`. However, changing the `package.json` to:
```
{
  "dependencies": {
    "jspm": "tamird/jspm-cli#dont-rewrite-package-json"
  }
}
```
and re-running `yarn install` does not change the contents of `node_modules` (it does update `yarn.lock`. Running `yarn cache clear && yarn install` does correctly update the contents of `node_modules`.

```
$ node --version
v7.0.0
$ yarn --version
0.16.1
$ uname -a
Darwin Tamirs-MBP 16.1.0 Darwin Kernel Version 16.1.0: Thu Oct 13 21:26:57 PDT 2016; root:xnu-3789.21.3~60/RELEASE_X86_64 x86_64
```

**Comment by markstos** [comment]
This may be a dupe of one of the other Github/caching issues already filed:

  * #1171
  * #1453
  * #1573

@tamird Could you look through other Github/caching issues including the above but possibly others (there are still more!) and confirm if you've really found a unique issue or if this is the same as one of the others? If it's a dupe, please close this copy of the issue, thanks.

**Comment by tamird** [comment]
It's possible that those issues are related, but none of them are direct
duplicates of this one (i.e. the root cause may be the same but the symptom
is different).

**Comment by bestander** [comment]
Should be fixed now #2074
